### ISSUE TITLE
Increase contrast of graph grids in high contrast mode

### ISSUE THREAD

**Issue by LucyGwilliamAdmin** [open-sdg/open-sdg#1094]
Graph grid lines are quite faint - may not even pass contrast guidelines:
![image](https://user-images.githubusercontent.com/43404358/108059988-31313300-704e-11eb-9cfd-04853d65a7c8.png)

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin @phillipgardiner 
While we are at this, should we also increase the contrast of lines in default mode? I am not necessarily arguing for that, but just bringing it up for discussion.

![indicator_1-1-1(1)](https://user-images.githubusercontent.com/1319083/108063146-c8838f80-7028-11eb-8818-060e4b8c1246.png)

Also, I mispoke on the call when I said it was the same color. Right now it is `#222222` for high contrast and `#dddddd` for default contrast. Any thoughts on what we should change to?

Good news is that it will be very easy to make the change by tweaking this line: https://github.com/open-sdg/open-sdg/blob/1.3.0-dev/_includes/assets/js/indicatorView.js#L536 :)

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning @phillipgardiner both of these fail the graphical objects and UI components contrast checks

DC:
![image](https://user-images.githubusercontent.com/43404358/108065760-5f1a7580-7056-11eb-8085-74f8b0e834d3.png)


HC:
![image](https://user-images.githubusercontent.com/43404358/108065719-50cc5980-7056-11eb-9082-cef4950b64af.png)

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin Cool - what would be the minimums needed to pass the "graphical objects AA" test?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning #949494 for default contrast and #5C5C5C for high contrast

**Comment by phillipgardiner** [comment]
@LucyGwilliamAdmin You just beat me to it! I only went away for 5 minutes to work it out and you've done it all :)

Just out of interest what is the # of the actual line on the graph?

**Comment by LucyGwilliamAdmin** [comment]
@phillipgardiner sorry, didn't realise you were also looking at it!

Do you mean the headline data line?

**Comment by phillipgardiner** [comment]
@LucyGwilliamAdmin its fine, its great you've done it. Yeah the Headline is it #777777?

**Comment by brockfanning** [comment]
Just put up #1095 if you would like to try out a feature branch.

I guess it is the headline color I was thinking of, which is identical in both default and high contrast.

**Comment by phillipgardiner** [comment]
@brockfanning Can the headline colour be a different colour in default and high contrast - as interesting this is something that has cropped up in some research I did recently (e.g. what should the default colour be)

**Comment by brockfanning** [comment]
I think that's one of those where I won't know how easy it is until I get in and try it out, but I have to assume that it is certainly possible.

**Comment by phillipgardiner** [comment]
@LucyGwilliamAdmin @brockfanning @AnnCorp  OK here are my thoughts. But unfortunately this is probably more work than what we originally thought! If we just change the high contrast grid lines, there is the potential that even though it passes accessibility guidelines, it actually become less usable as all the greys become quite similar.

For the high contrast could we consider setting the grid lines to grey (#5C5C5C) but change the headline data line to yellow (#FFFF00). This would make the default data line stand out from the rest of the graph whilst ensuring it all passes the contrast rules.

In default mode, during the last piece of research I did around disaggregation controls it was noted by participants that they were not always aware of the headline line (primarily due to it looking very similar in colour to the rest of the page - it didn't jump out). Do we therefore have an opportunity here to address this also? I have looked at the ONS site to see what they do and have used this as guidance (although one colour (orange) they use does not pass accessibility contrast rules!)

In default mode, could we consider changing the grid lines to grey (#949494) and the headline data line to blue (#004B70). This would have a knock on effect as we would also need to change the mouse hoover line to orange (#F36C12) to ensure it does not clash with the new headline data colour. 

Please note I have not considered how the new blue headline data line works against the other chart colours used to represent other dissagregations as I am not aware what these are. 

As I say this is just a proposal and happy to discuss. Unfortunatley it is not always possible to just change one thing without it having a knock on impact elsewhere.

**Comment by brockfanning** [comment]
For reference in this discussion, here are the default disaggregation colors (though it is configurable):

- ![#7e984f](https://www.thecolorapi.com/id?hex=7e984f&format=svg)
- ![#8d73ca](https://www.thecolorapi.com/id?hex=8d73ca&format=svg)
- ![#aaa533](https://www.thecolorapi.com/id?hex=aaa533&format=svg)
- ![#c65b8a](https://www.thecolorapi.com/id?hex=c65b8a&format=svg)
- ![#4aac8d](https://www.thecolorapi.com/id?hex=4aac8d&format=svg)
- ![#c95f44](https://www.thecolorapi.com/id?hex=c95f44&format=svg)

I'll follow up with some screenshots.

**Comment by brockfanning** [comment]
Here are screenshots of the suggestions above. Also, the code for these has been pushed to #1095 so a feature branch could be made if it would help.

![chart-colors-1](https://user-images.githubusercontent.com/1319083/108531080-4b644e80-72a4-11eb-94be-e80f9068fe26.png)

![chart-colors-2](https://user-images.githubusercontent.com/1319083/108531089-4ef7d580-72a4-11eb-9550-b9ed6f89eeda.png)

![chart-colors-3](https://user-images.githubusercontent.com/1319083/108531103-528b5c80-72a4-11eb-9889-18ae5ebfb35b.png)

![chart-colors-4](https://user-images.githubusercontent.com/1319083/108531111-55864d00-72a4-11eb-87a4-017ef75335c5.png)

**Comment by brockfanning** [comment]
Kind of a devil's advocate here but: even though I know we're using the minimum WCAG values there, I find my brain is getting vaguely confused by all the intersecting lines...

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning I see what you mean - it looks kind of cluttered now.

**Comment by phillipgardiner** [comment]
@brockfanning I don't think that is playing devils advocate at all, I had a feeling that may be an issue and totally agree. I'm just doing some research around 1) Do they have to pass contrast ratios as they are supplementary 2) Do we even need them at all 3) Do we only need the x-axis. Ultimately its a case of understanding why they exist in the first place and are they used, aswell as do they need to pass guidelines as I have not found an example yet where the grid lines are as dark as we propose. 

@LucyGwilliamAdmin is it possible to build the feature branch so I can see it in action please?

**Comment by phillipgardiner** [comment]
@brockfanning @LucyGwilliamAdmin Depending on all the above we may also have to look at the default dissagregation colours but I think that is a separate ticket (thanks for posting what they are :)

**Comment by phillipgardiner** [comment]
@brockfanning @LucyGwilliamAdmin Right done some digging. @LucyGwilliamAdmin this is possibly one to go back to DAC and double check. I think this article articulates my thinking pretty well (https://www.accessibility-developer-guide.com/knowledge/colours-and-contrast/graphical-objects/). 

In our specific instance we 'mark' each data point with a circle etc thus it is possible to interpret the graph without the grid lines (they are therefore supplementary and therefore could remain a colour outside of passing colour ratios). We also provide a 'table' option which again provides an alternative for interpretation if required. 

Given the above, I am therefore interpreting that we change the main x-axis and the y-axis lines to a dark grey (#757575) - apparently if something is less than 3 CSS pixels it has to have a contrast ration of 4:5:1 rather than 3:1. This grey would work for both the default and high contrast graphs.  Leave the rest of the grid lines as they were before. I don't think this would cause any accessibility issues (but like I say might be worth checking with DAC?)

@brockfanning On a side note, if and when there is any time. I would be really interested in seeing what the graphs would look like with only x-axis lines (x-axis only is what ONS seem to do on their main site) and seems the more common approach for our types of graphs and data type and the best practice. 

Thanks both, happy to discuss :)

**Comment by phillipgardiner** [comment]
@brockfanning The more I look into this, the more I am certain we do not need vertical grid lines (as we have the mouse hoover and the little circles at data points). Could we please mock up without them?

**Comment by brockfanning** [comment]
@phillipgardiner Sure thing, here is a an example of what that would look like - without any of the other changes above.

![screenshot-127 0 0 1_4000-2021 02 23-05_54_55](https://user-images.githubusercontent.com/1319083/108834195-dd69a100-759b-11eb-9f56-bd2eebf03a48.png)

**Comment by phillipgardiner** [comment]
@brockfanning thank you. That confirms it for me, we don't need the vertical lines :)

**Comment by phillipgardiner** [comment]
@brockfanning for clarity's sake I am posting the above comment again but with ALL final decisions in.

Given all of the above, I am therefore interpreting that we should:

- Change the main x-axis and the y-axis lines to a dark grey (#757575). This grey would work for both the default and high contrast graphs.

-Remove the y-axis grid lines. Display only x-axis grid lines. These x-axis will remain their current colour

- Update the default data line Blue colour to (#004466)

- Update the mouse hoover line to dark grey (#757575) on both default and high contrast.

- Update the mouse hover line to a dashed line

- Keep the high contrast default data line yellow (#FFFF00). However this will change once https://github.com/open-sdg/open-sdg/issues/1099 is implemented.

Any additional questions / clarity required please ask :)

**Comment by brockfanning** [comment]
Thanks @phillipgardiner - I believe I have all of this in #1095.
